tests/lib/rules/require-tagless-components.js

```diff
@@ -0,0 +1,124 @@
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/require-tagless-components');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE_REQUIRE_TAGLESS_COMPONENTS: ERROR_MESSAGE } = rule;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+    ecmaFeatures: {
+      legacyDecorators: true,
+    },
+  },
+});
+
+ruleTester.run('require-tagless-components', rule, {
+  valid: [
+    `
+      import Component from '@ember/component';
+      export default Component.extend({ tagName: '' });
+    `,
+    `
+      import Component from '@ember/component';
+      export default class MyComponent extends Component {
+        tagName = ''
+      }
+    `,
+    `
+      import Component from '@ember/component';
+      import { tagName } from '@ember-decorators/component';
+      @tagName('')
+      export default class MyComponent extends Component {}
+    `,
+    `
+      import Component from '@glimmer/component';
+      export default class MyComponent extends Component {}
+    `,
+    `
+      import SomeOtherThing from 'some-other-module';
+      export default SomeOtherThing.extend({
+        tagName: ''
+      });
+    `,
+    `
+      import SomeOtherThing from 'some-other-module';
+      export default class MyThing extends SomeOtherThing {
+        tagName = '';
```

Comment:
Wouldn't this test case be more useful if it had `tagName = 'some-tag'` since that's what the rule normally disallows? We need some valid examples testing to ensure that the aspects disallowed by the rule are allowed when inside a class that the rule doesn't apply to.

Reply:
Yes, absolutely! Oversight on my part, good catch!